dts/arm/nuvoton/npcx7.dtsi

```diff
@@ -78,6 +78,7 @@
 			interrupts = <33 3>;
 			clocks = <&pcc NPCX_CLOCK_BUS_APB2 NPCX_PWDWN_CTL1 4>;
 			pinctrl-0 = <&alta_uart1_sl1>; /* PIN10.11 */
+			pinctrl-names = "default";
```

Comment:
why is pin selection performed at the SoC DT files? This belongs to the board, the place where the selection is made based on board layout.

Reply:
Hey @gmarull,  thanks for your comments. The followings are my reasons:
 

- In Nuvoton npcx ec series, most peripheral devices have only one alternative function (ie. a one-to-one mapping between GPIO and itself.) and there are up to 120 IO pins. Putting plenty of `pinctrl-names = "default"` influences the board layout DT files which projects use npcx series significantly.

- From an application point of view, it is necessary to put `pinctrl-names` and relevant `pinctrl-0/1/2` properties in the board layout DT files if the users need to change the pinctrl selection dynamically.  For example, if users want to switch pin-mux from i2c to gpio and toggle SCL bus to recovery slave device, they can add the followings in board layout DT files. Then, use pinctrl api to change pin-mux dynamically. But for the single pinctrl selection case, no matter npcx or the other soc drivers apply default pinctrl state only. In this case, is it better to keep them in soc DT files and overwrite `pinctrl-names` and relevant `pinctrl-0/1/2` properties  in board layout DT files if needed?
```
&i2c0_0 {
        status = "okay";
        pinctrl-0 = <&alt2_i2c0_0_sl>;
        pinctrl-1 = <&alt2_no_i2c0_0_sl>;
        pinctrl-names = "default", "gpio";
};
```

Reply:
Sorry for late response. Please see my reply below.
> > Hey @gmarull, thanks for your comments. The followings are my reasons:
> > 
> > * In Nuvoton npcx ec series, most peripheral devices have only one alternative function (ie. a one-to-one mapping between GPIO and itself.) and there are up to 120 IO pins. Putting plenty of `pinctrl-names = "default"` influences the board layout DT files which projects use npcx series significantly.
> 
> This is how it is expected to work. In some cases it's not about pinmux only, but also pin properties such as pull-ups, etc. If you say that most of your SoC pins have a single working selection, then it may be ok to have them at SoC, provided boards can NEVER change any aspect of the configuration. If not, I don't think current method is acceptable.
> 

As @keith-zephyr mentioned, the functions such as pull-ups, open-drain, and so on are configured in `gpios` not `pinctrl-0/1/2` property in our projects. And we only use the single selection of pinctrl status in board layout DT files so far.

> > * From an application point of view, it is necessary to put `pinctrl-names` and relevant `pinctrl-0/1/2` properties in the board layout DT files if the users need to change the pinctrl selection dynamically.  For example, if users want to switch pin-mux from i2c to gpio and toggle SCL bus to recovery slave device, they can add the followings in board layout DT files. Then, use pinctrl api to change pin-mux dynamically.
> 
> Only drivers are expected to use pinctrl, not applications directly.

Got it. Since only drivers are expected to use pinctrl API, any attempt to change pinctrl status must be within the related driver. For example, if there is a need to unwedge i2c bus by toggling SCL bus, it is expected to add an i2c api called i2c_unwedge() and change related pin-mux status in its driver. Am I right?

Reply:
For the I2C bus recovery use case, the I2C controller driver could use pinctrl to switch to GPIO mode as part of it's `recover_bus` implementation.

Reply:
Ack